instrumentation/src/main/java/io/opentelemetry/android/features/diskbuffering/SignalDiskExporter.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.features.diskbuffering
+
+import androidx.annotation.WorkerThread
+import io.opentelemetry.contrib.disk.buffering.LogRecordDiskExporter
+import io.opentelemetry.contrib.disk.buffering.MetricDiskExporter
+import io.opentelemetry.contrib.disk.buffering.SpanDiskExporter
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+/**
+ * Entrypoint to read and export previously cached signals.
+ */
+class SignalDiskExporter internal constructor(
+    private val spanDiskExporter: SpanDiskExporter?,
+    private val metricDiskExporter: MetricDiskExporter?,
+    private val logRecordDiskExporter: LogRecordDiskExporter?,
+    private val exportTimeoutInMillis: Long,
```

Comment:
```suggestion
    private val exportTimeoutInMillis: Long,
    private val spanDiskExporter: SpanDiskExporter?,
    private val metricDiskExporter: MetricDiskExporter?,
    private val logRecordDiskExporter: LogRecordDiskExporter?,
```

Just personal preference, non optional params go first